src/firewall/nft.rs

```diff
@@ -0,0 +1,1114 @@
+                batch.add(make_rule(
+                    INPUTCHAIN,
+                    vec![
+                        get_subnet_match(&subnet, "saddr", stmt::Operator::EQ),
+                        stmt::Statement::Match(stmt::Match {
+                            left: expr::Expression::Named(expr::NamedExpression::Payload(
+                                expr::Payload {
+                                    protocol: "udp".to_string(),
+                                    field: "dport".to_string(),
+                                },
+                            )),
+                            right: expr::Expression::Number(53),
+                            op: stmt::Operator::EQ,
+                        }),
+                        stmt::Statement::Accept(None),
+                    ],
+                ));
+                // Forward chain: ip daddr <subnet> ct state related,established accept
+                batch.add(make_rule(
+                    FORWARDCHAIN,
+                    vec![
+                        get_subnet_match(&subnet, "daddr", stmt::Operator::EQ),
+                        stmt::Statement::Match(stmt::Match {
+                            left: expr::Expression::Named(expr::NamedExpression::CT(expr::CT {
+                                key: "state".to_string(),
+                                family: None,
+                                dir: None,
+                            })),
+                            right: expr::Expression::List(vec![
+                                expr::Expression::String("established".to_string()),
+                                expr::Expression::String("related".to_string()),
+                            ]),
+                            op: stmt::Operator::IN,
+                        }),
+                        stmt::Statement::Accept(None),
+                    ],
+                ));
+                // Forward chain: ip saddr <subnet> accept
+                batch.add(make_rule(
+                    FORWARDCHAIN,
+                    vec![
+                        get_subnet_match(&subnet, "saddr", stmt::Operator::EQ),
+                        stmt::Statement::Accept(None),
+                    ],
+                ));
+                // Postrouting chain: ip saddr <subnet> jump <chain>
+                batch.add(make_rule(
+                    POSTROUTINGCHAIN,
+                    vec![
+                        get_subnet_match(&subnet, "saddr", stmt::Operator::EQ),
+                        get_jump_action(&chain),
+                    ],
+                ));
+            }
+        }
+
+        let rules = batch.to_nftables();
+
+        helper::apply_ruleset(&rules, None, None)?;
+
+        Ok(())
+    }
+
+    fn teardown_network(&self, tear: internal_types::TearDownNetwork) -> NetavarkResult<()> {
+        if !tear.complete_teardown {
+            log::info!("Nothing to tear down, network still in use");
+
+            return Ok(());
+        }
```

Comment:
This is odd, the same check is already done in the bridge code so teardown_network() is never called in this case.
Reading the iptables code confuses me because there is bunch of conditionals on that that IMO really make no sense.

Anyway I think you can just remove the condition here.

Reply:
Should we just remove the `complete_teardown` bool entirely if this is the case?

Reply:
Ah, TearDownNetwork is not unique to firewall, so we can't. OK